config/config-api/src/com/thoughtworks/go/config/materials/MaterialConfigs.java

```diff
@@ -155,22 +149,43 @@ public void validate(ValidationContext validationContext) {
     }
 
     private void validateScmMaterials() {
-        List<ScmMaterialConfig> scmMaterials = filterScmMaterials();
-        int numberOfMaterials = scmMaterials.size();
-        for (ScmMaterialConfig material : scmMaterials) {
-            if (numberOfMaterials > 1) {
+        List<MaterialConfig> allSCMMaterials = getSCMAndPluggableSCMConfigs();
+        if (allSCMMaterials.size() > 1) {
+            for (MaterialConfig material : allSCMMaterials) {
                 if (StringUtil.isBlank(material.getFolder())) {
-                    material.setDestinationFolderError("Destination directory is required when specifying multiple scm materials");
+                    String fieldName = material instanceof ScmMaterialConfig ? ScmMaterialConfig.FOLDER : PluggableSCMMaterialConfig.FOLDER;
+                    material.addError(fieldName, "Destination directory is required when specifying multiple scm materials");
+                } else {
+                    validateDestinationFolder(allSCMMaterials, material);
                 }
-                for (ScmMaterialConfig otherMaterial : scmMaterials) {
-                    if (otherMaterial != material) {
-                        otherMaterial.validateNotSubdirectoryOf(material);
-                        otherMaterial.validateDestinationDirectoryName(material);
-                    }
+            }
+        }
+    }
+
+    private List<MaterialConfig> getSCMAndPluggableSCMConfigs() {
+        List<ScmMaterialConfig> scmMaterials = filterScmMaterials();
+        List<PluggableSCMMaterialConfig> pluggableSCMMaterials = filterPluggableSCMMaterials();
+        List<MaterialConfig> allSCMMaterials = new ArrayList<MaterialConfig>();
+        allSCMMaterials.addAll(scmMaterials);
+        allSCMMaterials.addAll(pluggableSCMMaterials);
+        return allSCMMaterials;
+    }
+
+    private void validateDestinationFolder(List<MaterialConfig> allSCMMaterials, MaterialConfig material) {
+        String materialFolder = material.getFolder();
+        for (MaterialConfig otherMaterial : allSCMMaterials) {
+            if (otherMaterial != material) {
+                if (otherMaterial instanceof ScmMaterialConfig) {
```

Comment:
Can we put all of these if-else ladders somewhere else? I'm getting worried that there are too many places this is being done. Putting them in one place is not very different, but it's at least very visible where it is being used.